extensions/gitpod-shared/src/features.ts

```diff
@@ -954,37 +951,10 @@ function createTaskPty(alias: string, context: GitpodExtensionContext, contextTo
 				await new Promise(resolve => setTimeout(resolve, 2000));
 			}
 		},
-		close: async () => {
-			if (token.isCancellationRequested) {
-				return;
-			}
-			tokenSource.cancel();
-
-			// await to make sure that close is not cause by the extension host process termination
-			// in such case we don't want to stop supervisor terminals
-			setTimeout(async () => {
-				if (contextToken.isCancellationRequested) {
-					return;
-				}
-				// Attempt to kill the pty, it may have already been killed at this
-				// point but we want to make sure
-				try {
-					const request = new ShutdownTerminalRequest();
-					request.setAlias(alias);
-					await util.promisify(context.supervisor.terminal.shutdown.bind(context.supervisor.terminal, request, context.supervisor.metadata, {
-						deadline: Date.now() + context.supervisor.deadlines.short
-					}))();
-					context.logger.trace(`${alias} terminal closed`);
-				} catch (e) {
-					if (e && e.code === grpc.status.NOT_FOUND) {
-						// Swallow, the pty has already been killed
-					} else {
-						context.logger.error(`${alias} terminal: shutdown failed:`, e);
-						console.error(`${alias} terminal: shutdown failed:`, e);
-					}
-				}
-			}, 1000);
-
+		close: () => {
+			// Do nothing as this gets fired when user kill terminal using UI
+			// and when closing the window.
+			// To really kill the terminal user must type `exit`
```

Comment:
yes, it will be a regression with closing :( I wonder whether there is another way to detect explicit closing by a user.

But we should also resolve user issues getting closed task terminals which is quite bad compare to introduced regression.